### PR TITLE
planner, executor: handle default frame for window functions

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 

When there is no frame specification, we need to add default frames to the window.

### What is changed and how it works?

- When there is no frame specification and the function needs frame,
    - if it has order by clause, we will use `RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW` as default frame.
    - Otherwise, we will leave it as empty, which is the same as `RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING`.

- For functions that cannot have frame but a frame specification presents, we will remove that frame and raise a note.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - None

Related changes

 - None

### CODE REVIEW

File: expression/aggregation/window_func.go [hunk 1]
```diff
@@ -27,3 +30,22 @@ type WindowFuncDesc struct {
 func NewWindowFuncDesc(ctx sessionctx.Context, name string, args []expression.Expression) *WindowFuncDesc {
 	return &WindowFuncDesc{newBaseFuncDesc(ctx, name, args)}
 }
+
+// partitionWindowFuncs is the functions that operate on the entire partition,
+// they should not have frame specifications.
+var partitionWindowFuncs = map[string]struct{}{
```
Comment (zz-jason) [inline expression/aggregation/window_func.go]: how about s/partitionWindowFuncs/noFrameWindowFuncs/?

File: planner/core/explain.go [hunk 2]
```diff
@@ -306,8 +306,8 @@ func (p *PhysicalWindow) formatFrameBound(buffer *bytes.Buffer, bound *FrameBoun
 	}
 	if bound.UnBounded {
 		buffer.WriteString("unbounded")
-	} else if bound.CalcFunc != nil {
-		sf := bound.CalcFunc.(*expression.ScalarFunction)
+	} else if len(bound.CalcFuncs) > 0 {
+		sf := bound.CalcFuncs[0].(*expression.ScalarFunction)
```
Comment (zz-jason) [inline planner/core/explain.go]: should we iterate all the `CalcFuncs`?

Comment (lamxTyler) [reply planner/core/explain.go]: No, the only case which we have multiple funcs is for `current row`, and it has already been handled.

**Pull request merged**